### PR TITLE
ANY23-426 Address Javadoc WARNING's 

### PR DESCRIPTION
**Pull request by lewismc**
This PR addresses https://issues.apache.org/jira/browse/ANY23-426

### CODE REVIEW

File: api/src/main/java/org/apache/any23/configuration/Setting.java [hunk 1]
```diff
@@ -113,6 +113,8 @@ public final Type getValueType() {
     }
 
     /**
+     * @param setting a setting key to retrieve
+     * @param <S> generic type for setting
```
Comment (HansBrende) [inline api/src/main/java/org/apache/any23/configuration/Setting.java]: Perhaps something more along the lines of:

```
@param setting a setting that may or may not have the same key as this setting
@param <S> the type of the supplied setting
```

File: api/src/main/java/org/apache/any23/configuration/Setting.java [hunk 2]
```diff
@@ -121,6 +123,7 @@ public final Type getValueType() {
     }
 
     /**
+     * @param newValue a value to provide for the new setting pair
```
Comment (HansBrende) [inline api/src/main/java/org/apache/any23/configuration/Setting.java]: Simply 
```
a value for a new setting
```
should suffice.

File: api/src/main/java/org/apache/any23/configuration/Settings.java [hunk 3]
```diff
@@ -132,7 +137,8 @@ public static Settings of(Setting<?>... settings) {
     }
 
     /**
-     * Returns a {@link Settings} object containing the supplied settings.
+     * @param c a polulated {@link java.util.Collection} of {@link org.apache.any23.configuration.Setting}'s
```
[the remark this answers is not in the document]
Comment (lewismc) [reply api/src/main/java/org/apache/any23/configuration/Settings.java]: Thanks

File: api/src/main/java/org/apache/any23/configuration/Setting.java (hunk 1 above)
Comment (lewismc) [reply api/src/main/java/org/apache/any23/configuration/Setting.java]: Thanks

File: api/src/main/java/org/apache/any23/configuration/Setting.java (hunk 2 above)
Comment (lewismc) [reply api/src/main/java/org/apache/any23/configuration/Setting.java]: Thanks

**Comment by lewismc** [comment]
Thanks for review @HansBrende

**Pull request merged**